crates/router/src/db/kafka_store.rs

```diff
@@ -1194,6 +1194,23 @@ impl MerchantConnectorAccountInterface for KafkaStore {
             .await
     }
 
+    #[cfg(all(feature = "oltp", feature = "v2"))]
+    async fn list_enabled_connector_accounts_by_profile_id(
+        &self,
+        state: &KeyManagerState,
+        profile_id: &id_type::ProfileId,
+        key_store: &domain::MerchantKeyStore,
+        connector_type: common_enums::ConnectorType,
+    ) -> CustomResult<Vec<domain::MerchantConnectorAccount>, errors::StorageError> {
+        self.list_enabled_connector_accounts_by_profile_id(
```

Comment:
```suggestion
        self.diesel_store.list_enabled_connector_accounts_by_profile_id(
```

Reply:
Good catch!
This would have caused infinite recursion otherwise!